### PR TITLE
2.0 Flat configuration and custom configuration 

### PR DESCRIPTION
**Pull request by pepibumur**
Related: https://github.com/tuist/tuist/pull/3374

### Short description 📝
I'm flattening `CustomConfiguration` and `Configuration` into a single model, `Configuration` for the sake of simplicity.

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [ ] In case the PR introduces changes that affect users, the documentation has been updated.

### CODE REVIEW

File: Sources/ProjectDescription/Settings.swift [hunk 1]
```diff
@@ -55,66 +55,56 @@ public enum SettingValue: ExpressibleByStringInterpolation, ExpressibleByArrayLi
...
 ///
 /// - seealso: Configuration
-public struct CustomConfiguration: Equatable, Codable {
+public struct Configuration: Equatable, Codable {
     public enum Variant: String, Codable {
         case debug
         case release
     }
 
     public let name: String
```
Comment (kwridan) [inline Sources/ProjectDescription/Settings.swift:70]: We can possibly leverage `ConfigurationName` here too to allow users to have a shared type for this.

e.g.

```swift
extension ConfigurationName {
  static var beta: ConfigurationName {
     .configuration("Beta")
  }
}

let configurations: [Configuration] = [
  .release(name: .debug, xcconfig: "Configs/debug.xcconfig"), 
  .release(name: .beta, xcconfig: "Configs/beta.xcconfig"),
  .release(name: .release, xcconfig: "Configs/debug.xcconfig"),
]

let project = Project(
  name: "MyMultiConfigProject",
  settings: .settings(configurations: configurations),
 // ...
)
```

The same type can then be used in the other APIs that require only the configuration by name, e.g. in `RunAction`, `TestAction` etc...

**Review by kwridan — COMMENTED** [review]
Thanks @pepibumur 

Happy to assist with this change

File: Sources/TuistLoaderTesting/Loaders/TestData/ProjectDescription+TestData.swift [hunk 2]
```diff
@@ -142,7 +142,7 @@ extension TestAction {
     public static func test(targets: [TestableTarget] = [],
                             arguments: Arguments? = nil,
                             configuration: ConfigurationName = .debug,
-                            coverage: Bool = true) -> TestAction
+                            coverage _: Bool = true) -> TestAction
```
Comment (danyf90) [inline Sources/TuistLoaderTesting/Loaders/TestData/ProjectDescription+TestData.swift:145]: This should be passed to `TestAction.targets` below

**Review by danyf90 — APPROVED** [review]

File: Sources/TuistLoaderTesting/Loaders/TestData/ProjectDescription+TestData.swift (hunk 2 above)
Comment (kwridan) [reply Sources/TuistLoaderTesting/Loaders/TestData/ProjectDescription+TestData.swift:145]: ah good spot we missed this!

https://github.com/tuist/tuist/pull/3520

**Pull request merged**